cuda_core/cuda/core/experimental/_program.py

```diff
@@ -422,7 +422,26 @@ def __post_init__(self):
         if self.numba_debug:
             self._formatted_options.append("--numba-debug")
 
-    def _as_bytes(self):
+    def as_bytes(self) -> list[bytes]:
+        """Convert the formatted program options to a list of byte strings.
+
+        This method encodes each of the formatted options stored in this
+        `ProgramOptions` instance into byte strings, suitable for passing
+        to C libraries that perform NVRTC compilation.
```

Comment:
`Program` wraps multiple libraries so I'd like to keep it vague. 
```suggestion
        to C libraries that calls the underlying compiler library.
```